catt/http_server.py

```diff
@@ -1,41 +1,41 @@
-import os
 import socketserver
 import time
 import traceback
 from http.server import BaseHTTPRequestHandler
+from pathlib import Path
 
 
 def serve_file(filename, address="", port=45114):
     class FileHandler(BaseHTTPRequestHandler):
         def do_GET(self):  # noqa
             try:
-                file = open(filename, "rb")
-                stat = os.fstat(file.fileno())
-                length = stat.st_size
+                mediapath = Path(filename)
+                length = mediapath.stat().st_size
+                mtime = mediapath.stat().st_mtime
+                mediafile = open(str(mediapath), "rb")
 
                 self.send_response(200)
                 self.send_header("Content-type", "video/mp4")
                 self.send_header("Content-Length", length)
-                self.send_header("Accept-Ranges", "bytes")
                 self.send_header(
                     "Last-Modified",
                     time.strftime(
                         "%a %d %b %Y %H:%M:%S GMT",
-                        time.localtime(os.path.getmtime(filename))
+                        time.localtime(mtime)
                     )
                 )
                 self.end_headers()
 
                 while True:
-                    data = file.read(100 * 1024)
+                    data = mediafile.read(100 * 1024)
```

Comment:
It does not appear to be possible to do this with ```pathlib```.

Reply:
What isn't possible, you mean?

Reply:
Partial reading, which is why I used built-in ```open()```.

Reply:
Ah, yes, unfortunately the `read()` parameter doesn't take ranges.